src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php

```diff
@@ -97,6 +97,10 @@ public function load(array $configs, ContainerBuilder $container)
             if (is_dir($dir = $bundle['path'].'/Resources/views')) {
                 $this->addTwigPath($twigFilesystemLoaderDefinition, $dir, $name);
             }
+
+            if (null !== $bundle['parent']) {
+                $this->prependParentPaths($twigFilesystemLoaderDefinition, $container, $name, $bundle['parent']);
```

Comment:
Prepending paths looks wrong to me here, as it would overwrite user-configured paths for the same namespace too.
What you should do is collecting the child bundles for each bundle, and then adding paths for child bundles before adding the path of the bundle itself